packages/opds-web-client/src/components/context/__tests__/PathForContextTest.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import { expect } from "chai";
+import { stub } from "sinon";
+
+import * as PropTypes from "prop-types";
+import { shallow, mount } from "enzyme";
+
+import PathForProvider, { PathForContext } from "../PathForContext";
+import { PathFor } from "../../../interfaces";
+import { isNull } from "util";
+
+const pathFor: PathFor = (collectionUrl, bookUrl) => {
+  return "path";
+};
+
+const WrappedPathForProvider = ({ children }) => {
+  return <PathForProvider pathFor={pathFor}>{children}</PathForProvider>;
+};
+
+describe("PathForContext", () => {
+  it("passes pathFor down via legacy API", () => {
+    /**
+     * unfortunately enzyme doesn't provide a clear way to test for
+     * access to context via legacy or new api that I can find.
+     * This is a somewhat hacky way to do it by testing for the context
+     * in a custom child component and then putting text in the dom
+     * which we will expect() on via enzyme. Not pretty, but it works.
+     */
+
+    const hasAccessProof = "does have access to context";
+
+    class Child extends React.Component {
+      static contextTypes = {
+        pathFor: PropTypes.func.isRequired
+      };
+      render() {
+        // here is the meat of the test
+        const hasContext = this.context?.pathFor === pathFor;
```

Comment:
What is the `?` after `this.context` for? I haven't seen it be used this way before.

Reply:
It's part of typescript 3.7 called optional chaining. Basically it makes it safe to access deeply nested properties without having to test for existence at every level. So if `this.context` is undefined it won't spit out a type error and will instead evaluate to undefined. See more here https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Reply:
That is amazing and really great. Thanks! That's something @adriana-alter and I can use now.

Reply:
It has changed my life already 🙏 !